src/torcontrol.cpp

```diff
@@ -326,6 +326,53 @@ TorController::~TorController()
     }
 }
 
+void TorController::get_socks_cb(TorControlConnection& _conn, const TorControlReply& reply)
+{
+    // NOTE: We can only get here if -onion is unset
+    std::string socks_location;
+    if (reply.code == 250) {
+        for (const auto& line : reply.lines) {
+            if (0 == line.compare(0, 20, "net/listeners/socks=")) {
+                const std::string port_list_str = line.substr(20);
+                std::vector<std::string> port_list;
+                boost::split(port_list, port_list_str, boost::is_any_of(" "));
+                for (auto& portstr : port_list) {
+                    if (portstr.empty()) continue;
+                    if ((portstr[0] == '"' || portstr[0] == '\'') && (*portstr.rbegin() == portstr[0])) {
+                        portstr = portstr.substr(1, portstr.size() - 2);
+                        if (portstr.empty()) continue;
+                    }
+                    socks_location = portstr;
+                    if (0 == portstr.compare(0, 10, "127.0.0.1:")) {
+                        // Prefer localhost - ignore other ports
+                        break;
+                    }
+                }
+            }
+        }
+        if (!socks_location.empty()) {
+            LogPrint(BCLog::TOR, "tor: Get SOCKS port command yielded %s\n", socks_location);
+        } else {
+            LogPrintf("tor: Get SOCKS port command returned nothing\n");
+        }
+    } else if (reply.code == 510) {  // 510 Unrecognized command
+        LogPrintf("tor: Get SOCKS port command failed with unrecognized command (You probably should upgrade Tor)\n");
+    } else {
+        LogPrintf("tor: Get SOCKS port command failed; error code %d\n", reply.code);
+    }
+
+    if (socks_location.empty()) {
+        // Fallback to old behaviour
+        socks_location = "127.0.0.1";
+    }
+
+    CService resolved(LookupNumeric(socks_location, 9050));
```

Comment:
> But anyway - we can't expect the automatic configuration to detect and work in all cases. For this we have the manual override. @lano1106, in your case you can set `-onion=192.168.1.2:9050` which will work with `master` and with this PR.

you are correct but see my other suggestion below. I did configure the onion proxy with the following options:
-proxy=192.168.1.2:9050 -torcontrol=127.0.0.1:9151

I think that the condition:
`gArgs.GetArg("-onion", "") == ""`

should be changed with:
`!GetProxy(NET_ONION, addrOnion)`

because -onion switch is not the only way to configure the onion proxy. This is the root cause of my problem and why I got interested in this issue in the first place...

Reply:
Fixed (without calling LookupNumeric with an empty string)